### PR TITLE
Added source-maps support

### PR DESCRIPTION
**Pull request by iMoritz**
I added support for source-maps using the node-sass sourceMap option (#33)

### CODE REVIEW

**Comment by iMoritz** [comment]
Travis builds working fine on my forked repo :confused: https://travis-ci.org/iMoritz/node-sass-middleware/builds/63678281

File: middleware.js [hunk 1]
```diff
@@ -149,6 +163,20 @@ module.exports = function(options) {
                 }
               });
             });
+
+            if (sourceMap != null) {
```
Comment (am11) [inline middleware.js]: `if (sourceMap)`

File: middleware.js [hunk 2]
```diff
@@ -108,6 +110,12 @@ module.exports = function(options) {
         sassDir = dirname(sassPath);
       }
 
+      if (sourceMap === true) {
+        var cssMapPath = cssPath + '.map';
+      } else if (typeof sourceMap === 'string') {
+        var cssMapPath = join(dirname(cssPath), sourceMap);
+      }
+
```
Comment (am11) [inline middleware.js]: These should be passed to node-sass as is. We are processing the options there: [index.js#L66L74](https://github.com/sass/node-sass/blob/146c8e096e1797f656ad81e980e88a363d2eaba4/lib/index.js#L66L74).

**Comment by am11** [comment]
Thanks for the PR @iMoritz! :+1: 
LGTM modulo few nits.

@kevva, PTAL.

File: middleware.js [hunk 3]
```diff
@@ -163,19 +191,21 @@ module.exports = function(options) {
       var compile = function() {
         if (debug) { log('read', cssPath); }
 
-        fs.readFile(sassPath, 'utf8', function(err, str) {
-          if (err) {
-            error(err);
+        fs.exists(sassPath, function(exists) {
+          if (!exists) {
             return next();
+          } else {
```
Comment (kevva) [inline middleware.js]: You don't need `else` after `return`.

File: test/middleware.js [hunk 4]
```diff
@@ -187,6 +194,44 @@ describe('Using middleware to compile .scss', function () {
...
+        dest: __dirname,
+        sourceMap: true
+      }))
+      .use(function(err, req, res, next) {
+        res.statusCode = 500;
+        res.end(err.message);
+      });
+
+    it('generates source-map with correct contents', function(done) {
+
```
Comment (kevva) [inline test/middleware.js]: Remove this empty line.

File: middleware.js [hunk 5]
```diff
@@ -108,6 +110,12 @@ module.exports = function(options) {
         sassDir = dirname(sassPath);
       }
 
+      if (sourceMap === true) {
+        var cssMapPath = cssPath + '.map';
```
Comment (kevva) [inline middleware.js]: `var cssMapPath` should be defined outside the `if` statement.

File: test/middleware.js [hunk 6]
```diff
@@ -376,6 +427,45 @@ describe('Using middleware to compile .sass', function () {
...
+        indentedSyntax: true,
+        sourceMap: true
+      }))
+      .use(function(err, req, res, next) {
+        res.statusCode = 500;
+        res.end(err.message);
+      });
+
+    it('generates source-map with correct contents', function(done) {
+
```
Comment (kevva) [inline test/middleware.js]: Remove this empty line.

File: test/middleware.js [hunk 7]
```diff
@@ -376,6 +427,45 @@ describe('Using middleware to compile .sass', function () {
...
+                if (cont === result.map.toString()) {
+                  done();
+                }
+              } else {
+                setTimeout(checkFile, 10);
+              }
+            });
+          }());
+        });
+
```
Comment (kevva) [inline test/middleware.js]: Remove this empty line.

File: test/middleware.js [hunk 8]
```diff
@@ -376,6 +427,45 @@ describe('Using middleware to compile .sass', function () {
...
+    });
+
```
Comment (kevva) [inline test/middleware.js]: Remove this empty line.

File: test/middleware.js [hunk 9]
```diff
@@ -376,6 +427,45 @@ describe('Using middleware to compile .sass', function () {
...
+        .expect(200, function() {
+          var filesrc = fs.readFileSync(index_scssFile),
+              result = sass.renderSync({ file: index_sassFile, indentedSyntax: true, outFile: index_cssFile, sourceMap: true });
+
+          (function checkFile() {
+            fs.exists(index_sourceMap, function(exists) {
+              if (exists) {
+                var cont = fs.readFileSync(index_sourceMap).toString();
+                if (cont === result.map.toString()) {
+                  done();
```
Comment (kevva) [inline test/middleware.js]:
```js
return done();
```

File: test/middleware.js [hunk 10]
```diff
@@ -376,6 +427,45 @@ describe('Using middleware to compile .sass', function () {
...
+                }
+              } else {
```
Comment (kevva) [inline test/middleware.js]: Remove this `else` statement after you've added the `return` to the above.

**Comment by kevva** [comment]
Looks good apart from my comments.

File: middleware.js (hunk 2 above)
Comment (iMoritz) [inline middleware.js]: I need this path to write the file to the correct directory

Comment (am11) [inline middleware.js]: Ops, I thought you are setting this for upstream option.
Disregard my previous comment.

File: test/middleware.js (hunk 6 above)
Comment (iMoritz) [inline test/middleware.js]: which empty lines do you mean?

**Comment by am11** [comment]
Travis build was probably timing out. I restarted the job and it's green now.

File: test/middleware.js (hunk 6 above)
Comment (am11) [inline test/middleware.js]: Line 444

Comment (iMoritz) [inline test/middleware.js]: kk

File: middleware.js (hunk 2 above)
Comment (am11) [inline middleware.js]: You can get the *resolved* path by calling `this.options.sourceMap`.

File: middleware.js [hunk 11]
```diff
@@ -149,6 +164,20 @@ module.exports = function(options) {
...
               });
             });
+
+            if (sourceMap) {
+              mkdirp(dirname(cssMapPath), '0700', function(err) {
+                if (err) {
+                  return error(err);
+                }
+
+                fs.writeFile(cssMapPath, result.map, 'utf8', function(err) {
```
Comment (am11) [inline middleware.js]: `fs.writeFile(this.options.sourceMap, result.map, 'utf8', function(err))`
after that, you would not need `cssMapPath`.

**Comment by am11** [comment]
Thanks @iMoritz ! :tada: :star2:

**Pull request merged**